src/fparser/two/symbol_table.py

```diff
@@ -357,72 +575,47 @@ def add_data_symbol(self, name, primitive_type):
                     f"Symbol table already contains a use of a "
                     f"module with name '{name}'"
                 )
-            for mod_name, var_list in self._modules.items():
-                if var_list and lname in var_list:
+            for mod_name, mod in self._modules.items():
+                if mod.symbol_names and lname in mod.symbol_names:
                     raise SymbolTableError(
                         f"Symbol table already contains a use of a symbol "
                         f"named '{name}' from module '{mod_name}'"
                     )
 
         self._data_symbols[lname] = SymbolTable.Symbol(lname, primitive_type.lower())
 
-    def add_use_symbols(self, name, only_list=None):
+    def add_use_symbols(self, name, only_list=None, rename_list=None):
         """
         Creates an entry in the table for the USE of a module with the supplied
         name. If no `only_list` is supplied then this USE represents a wildcard
         import of all public symbols in the named module. If the USE statement
         has an ONLY clause but without any named symbols then `only_list`
         should be an empty list.
 
+        A USE can also have one or more rename entries *without* an only list.
+
         :param str name: the name of the module being imported via a USE. Not \
             case sensitive.
-        :param only_list: Whether or not there is an 'only:' clause on the \
-            USE statement and, if so, the names of the symbols being imported \
-            (not case sensitive).
-        :type only_list: NoneType or list of str
+        :param only_list: if there is an 'only:' clause on the USE statement \
+            then this contains a list of tuples, each holding the local name \
+            of the symbol and its name in the module from which it is \
+            imported. These names are case insensitive.
+        :type only_list: Optional[List[Tuple[str, str | NoneType]]]
```

Comment:
Here, and in the ModuleUse constructor, only_list is a list as is rename_list, but the properties return sets.

Reply:
Yes, fixed now.